### PR TITLE
Update ServiceRuntime proxy in Rust to the current Exonum revision [ECR-3698]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

Updated to revision 47114e3 as no functional changes were made since 0f44d1c

---
See: https://jira.bf.local/browse/ECR-3698

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/rust/build.rs [hunk 1]
```diff
@@ -12,5 +13,8 @@ fn main() {
         println!("cargo:rustc-link-lib=dylib=c++");
     }
 
-    protobuf_generate("../src/main/proto", &["src/proto"], "protobuf_mod.rs");
+    ProtobufGenerator::with_mod_name("protobuf_mod.rs")
+        .with_input_dir("../src/main/proto")
+        .add_path("../src/main/proto")
```
Comment (vitvakatu) [inline exonum-java-binding/core/rust/build.rs]: Do not even ask me about what is going on here and why we need two identical arguments for different methods.

File: exonum-java-binding/core/rust/src/proxy/runtime.rs [hunk 2]
```diff
@@ -452,6 +432,21 @@ impl Runtime for JavaRuntimeProxy {
             Ok(())
         }));
     }
+
+    fn shutdown(&self) {
+        unwrap_jni(self.exec.with_attached(|env| {
+            panic_on_exception(
```
[the remark this answers is not in the document]
Comment (vitvakatu) [reply exonum-java-binding/core/rust/src/proxy/runtime.rs]: Nope :) Created ECR-3725

File: exonum-java-binding/core/rust/src/utils/jni_cache.rs [hunk 3]
```diff
@@ -215,21 +214,15 @@ pub mod runtime_adapter {
     }
 
     /// Returns cached `JMethodID` for `ServiceRuntimeAdapter.createService()`.
```
Comment (skletsun) [inline exonum-java-binding/core/rust/src/utils/jni_cache.rs]: NIT: Outdated comment.

File: exonum-java-binding/core/rust/src/utils/jni_cache.rs [hunk 4]
```diff
@@ -215,21 +214,15 @@ pub mod runtime_adapter {
...
-    pub fn create_service_id() -> JMethodID<'static> {
+    pub fn restart_service_id() -> JMethodID<'static> {
         check_cache_initialized();
-        unsafe { RUNTIME_ADAPTER_CREATE_SERVICE.unwrap() }
+        unsafe { RUNTIME_ADAPTER_RESTART_SERVICE.unwrap() }
     }
 
     /// Returns cached `JMethodID` for `ServiceRuntimeAdapter.initializeService()`.
-    pub fn initialize_service_id() -> JMethodID<'static> {
+    pub fn add_service_id() -> JMethodID<'static> {
```
Comment (skletsun) [inline exonum-java-binding/core/rust/src/utils/jni_cache.rs]: NIT: The same.

**Pull request merged**